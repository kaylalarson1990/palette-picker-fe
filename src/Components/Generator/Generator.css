h1 {
  font-weight: 300;
  font-size: 50px;
}

p {
  color: black;
  font-size: 16px;
  background-color: rgba(255, 255, 255, 0.5);
  border-radius: 10px;
  padding: 5px;
}

.palette-container {
  display: flex;
  justify-content: space-around;
  margin: 5%;
}

.color-container {
  align-items: center;
  width: 20%;
  height: 600px;
  display: flex;
  flex-direction: column;
  justify-content: flex-end;
}

.color-lock {
  cursor: pointer;
  padding: 15px;
  border-radius: 10px;
  width: 40px;
  background-color: rgba(255, 255, 255, 0.5);
}

.color-lock:hover {
  width: 45px;
}

.generate-palette-btn,
.toggle-btn {
  padding: 10px;
  background-color: white;
  border: 1px solid #4D63AD;
  color: #4D63AD;
  font-size: 20px;
  border-radius: 5px;
  width: 300px;
  cursor: pointer;
  display: inline-block;
  letter-spacing: 1px;
  cursor: pointer;
  box-shadow: inset 0 0 0 0 #89B6E3;
  -webkit-transition: ease-out 0.4s;
  -moz-transition: ease-out 0.4s;
  transition: ease-out 0.4s;
}

.toggle-btn:hover {
  background-color: #4D63AD;
  color: white;
  box-shadow: inset 400px 0 0 0 #4D63AD;
}

.toggle-btn {
  width: 200px;
  margin: 10px;
}

.generate-palette-btn {
  background-color: #4D63AD;
  border: 1px solid #4D63AD;
  color: white;
}

.generate-palette-btn:hover {
  background-color: #89B6E3;
  border: 1px solid #89B6E3;
  color: white;
}

