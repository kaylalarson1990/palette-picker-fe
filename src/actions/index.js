export const gatherProjects = projects => ({
  type: 'GATHER_PROJECTS',
  projects
})

export const gatherPalettes = palettes => ({
  type: 'GATHER_PALETTES',
  palettes
})

export const gatherProject = id => ({
  type: 'GATHER_PROJECT',
  id
})

export const gatherPalette = id => ({
  type: 'GATHER_PALETTE',
  id
})