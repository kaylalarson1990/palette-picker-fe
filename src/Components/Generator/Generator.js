import React, { Component } from "react";
import randomHexColor from "random-hex-color";
import "./Generator.css";

class Generator extends Component {
  constructor(props) {
    super(props);
    this.state = {
      colors: ["", "", "", "", ""],
      isLocked: [false, false, false, false, false]
    };
  }

  componentDidMount() {
    this.generateHexColors();
  }

  checkIfLocked = (index) => {
    if (this.state.isLocked[index]) {
      return require("../../images/locked.svg");
    } else {
      return require("../../images/unlocked.svg");
    }
  };

  lockAHexColor = palette => {
    const { isLocked } = this.state;
    const newLocked = isLocked.map((bool, index) => {
      if (index === palette) {
        return !bool;
      } else {
        return bool;
      }
    });
    this.setState({ isLocked: newLocked });
  };


  generateHexColors = () => {
    const { isLocked, colors } = this.state;
    const newColors = colors.map((color, index) => {
      return isLocked[index] ? color : randomHexColor();
    });
    this.setState({ colors: newColors });
  };

  render() {
    const { colors } = this.state;
    let lockOne = this.checkIfLocked(0);
    let lockTwo = this.checkIfLocked(1);
    let lockThree = this.checkIfLocked(2);
    let lockFour = this.checkIfLocked(3);
    let lockFive = this.checkIfLocked(4);

    return (
      <div>
        <h1>Palette Picker</h1>
        <section class="palette-container">
          <div
            id="color1"
            class="color-container"
            style={{ backgroundColor: colors[0] }}
          >
            <img
              src={lockOne}
              id="1"
              class="color-lock"
              value="Lock Color"
              onClick={() => this.lockAHexColor(0)}
            />
            <p class="hex-color">{colors[0]}</p>
          </div>
          <div
            id="color2"
            class="color-container"
            style={{ backgroundColor: colors[1] }}
          >
            <img
              src={lockTwo}
              id="2"
              class="color-lock"
              value="Lock Color"
              onClick={() => this.lockAHexColor(1)}
            />
            <p class="hex-color">{colors[1]}</p>
          </div>
          <div
            id="color3"
            class="color-container"
            style={{ backgroundColor: colors[2] }}
          >
            <img
              src={lockThree}
              id="3"
              class="color-lock"
              value="Lock Color"
              onClick={() => this.lockAHexColor(2)}
            />
            <p class="hex-color">{colors[2]}</p>
          </div>
          <div
            id="color4"
            class="color-container"
            style={{ backgroundColor: colors[3] }}
          >
            <img
              src={lockFour}
              id="4"
              class="color-lock"
              value="Lock Color"
              onClick={() => this.lockAHexColor(3)}
            />
            <p class="hex-color">{colors[3]}</p>
          </div>
          <div
            id="color5"
            class="color-container"
            style={{ backgroundColor: colors[4] }}
          >
            <img
              src={lockFive}
              id="5"
              class="color-lock"
              value="Lock Color"
              onClick={() => this.lockAHexColor(4)}
            />
            <p class="hex-color">{colors[4]}</p>
          </div>
        </section>
        <section class="generate-palette">
          <input
            type="button"
            class="generate-palette-btn button"
            value="Click me to see more colors"
            onClick={this.generateHexColors}
          />
        </section>
        {/* <NewPaletteForm colors={this.state.colors} /> */}
      </div>
    );
  }
}

export default Generator;
