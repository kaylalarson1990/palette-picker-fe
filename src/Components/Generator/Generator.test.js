import React from "react";
import { Generator, mapDispatchToProps } from "./Generator";
import { gatherPalettes } from "../../actions/index";
import { shallow } from "enzyme";
import "../../../setupTests";

describe("Generator", () => {
  let wrapper, instance;
  beforeEach(() => {
    wrapper = shallow(<Generator />);
    instance = wrapper.instance();
  });

  it("should match the snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });
});

describe("palette lock (lockAHexColor)", () => {
  let wrapper, instance;
  beforeEach(() => {
    wrapper = shallow(<Generator />);
    instance = wrapper.instance();
  });
  it("should update the locked colors in state", () => {
    expect(wrapper.state("isLocked")).toEqual([
      false,
      false,
      false,
      false,
      false
    ]);
    instance.lockAHexColor(0);
    expect(wrapper.state("isLocked")).toEqual([
      true,
      false,
      false,
      false,
      false
    ]);
  });
});


describe("mapDispatchToProps", () => {
  it("calls dispatch with a gatherPalettes action", () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = gatherPalettes("Test Project");
    const mappedDispatch = mapDispatchToProps(mockDispatch);
    mappedDispatch.gatherPalettes("Test Project");
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});
