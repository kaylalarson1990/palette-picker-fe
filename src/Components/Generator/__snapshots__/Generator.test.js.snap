// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Generator should match the snapshot 1`] = `
<div>
  <h1>
    Color Therapy
  </h1>
  <section
    className="palette-container"
  >
    <div
      className="color-container"
      id="color1"
      style={
        Object {
          "backgroundColor": "#f7c269",
        }
      }
    >
      <img
        className="color-lock"
        id="1"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "unlocked.svg",
          }
        }
        value="Lock Color"
      />
      <p
        className="hex-color"
      >
        #f7c269
      </p>
    </div>
    <div
      className="color-container"
      id="color2"
      style={
        Object {
          "backgroundColor": "#8859d2",
        }
      }
    >
      <img
        className="color-lock"
        id="2"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "unlocked.svg",
          }
        }
        value="Lock Color"
      />
      <p
        className="hex-color"
      >
        #8859d2
      </p>
    </div>
    <div
      className="color-container"
      id="color3"
      style={
        Object {
          "backgroundColor": "#f83dda",
        }
      }
    >
      <img
        className="color-lock"
        id="3"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "unlocked.svg",
          }
        }
        value="Lock Color"
      />
      <p
        className="hex-color"
      >
        #f83dda
      </p>
    </div>
    <div
      className="color-container"
      id="color4"
      style={
        Object {
          "backgroundColor": "#0e5aaa",
        }
      }
    >
      <img
        className="color-lock"
        id="4"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "unlocked.svg",
          }
        }
        value="Lock Color"
      />
      <p
        className="hex-color"
      >
        #0e5aaa
      </p>
    </div>
    <div
      className="color-container"
      id="color5"
      style={
        Object {
          "backgroundColor": "#568c7a",
        }
      }
    >
      <img
        className="color-lock"
        id="5"
        onClick={[Function]}
        src={
          Object {
            "ReactComponent": Object {
              "$$typeof": Symbol(react.forward_ref),
              "render": [Function],
            },
            "__esModule": true,
            "default": "unlocked.svg",
          }
        }
        value="Lock Color"
      />
      <p
        className="hex-color"
      >
        #568c7a
      </p>
    </div>
  </section>
  <section
    className="generate-palette"
  >
    <input
      className="generate-palette-btn button"
      onClick={[Function]}
      type="button"
      value="Generate colors"
    />
  </section>
  <Connect(NewPaletteForm)
    addNewPalette={[Function]}
  />
</div>
`;
